### PR TITLE
Adding support cast rows to structs with embedded types

### LINKED ISSUE #84: CastRow does not support embedded types
# Overview

`CastRow` does not support embedded types. It simply ignores them.

cc/ @joeberth

---

Please preserve this line to notify @danielfireman (lead of this repository)

### PR DESCRIPTION
**Pull request by danielfireman**
# Overview

Fixes #84 

---

Please preserve this line to notify @danielfireman (lead of this repository)

---
This change is [Reviewable](https://reviewable.io/reviews/frictionlessdata/tableschema-go/86)

### CODE REVIEW

**Comment by danielfireman** [comment]
cc/ @joeberth

File: schema/schema.go [hunk 1]
```diff
@@ -198,36 +198,70 @@ func (s *Schema) CastRow(row []string, out interface{}) error {
 	if len(row) != len(s.Fields) {
 		return fmt.Errorf("The row with %d values does not match the %d fields in the schema", len(row), len(s.Fields))
 	}
+	fields, err := getStructFields(out)
+	if err != nil {
+		return fmt.Errorf("Errror extracting field information from the struct:%q", err)
```
Comment (ABuarque) [inline schema/schema.go:203]:
```suggestion
		return fmt.Errorf("error extracting field information from the struct:%q", err)
```

File: schema/schema.go [hunk 2]
```diff
@@ -198,36 +198,70 @@ func (s *Schema) CastRow(row []string, out interface{}) error {
 	if len(row) != len(s.Fields) {
 		return fmt.Errorf("The row with %d values does not match the %d fields in the schema", len(row), len(s.Fields))
```
Comment (ABuarque) [inline schema/schema.go:199]:
```suggestion
		return fmt.Errorf("the row with %d values does not match the %d fields in the schema", len(row), len(s.Fields))
```

File: schema/schema.go [hunk 3]
```diff
@@ -198,36 +198,70 @@ func (s *Schema) CastRow(row []string, out interface{}) error {
...
+type structField struct {
+	reflect.StructField
+	value reflect.Value
+}
+
+func (s *structField) Set(v interface{}) error {
+	toSetValue := reflect.ValueOf(v)
+	toSetType := toSetValue.Type()
+	if !toSetType.ConvertibleTo(s.Type) {
+		return fmt.Errorf("field:%s value:%v - can not convert from %v to %v", s.Name, v, toSetType, s.Type)
```
Comment (ABuarque) [inline schema/schema.go:237]:
```suggestion
		return fmt.Errorf("field:%s value:%v - cannot convert from %v to %v", s.Name, v, toSetType, s.Type)
```

**Pull request merged**